src/exo/API_scheduling.py

```diff
@@ -2056,12 +2362,17 @@ def fuse(proc, stmt1, stmt2, unsafe_disable_check=False):
 @sched_op([ForCursorA])
 def remove_loop(proc, loop_cursor):
     """
-    Remove the loop around some block of statements.
-    This operation is allowable when the block of statements in question
-    can be proven to be idempotent.
+    Remove the loop around some block of statements. This operation is allowable
+    when the block of statements in question can be proven to be idempotent. If the
+    loop is not guaranteed to execute at least once, we wrap a guard around its body.
 
     args:
         loop_cursor     - cursor pointing to the loop to remove
+        rc              - (bool) whether to return relevant cursors
+
+    relevant cursors:
+        block           - a cursor to the block of statements that was inside the loop
+        ?guard          - a cursor to the guard around the loop body
```

Comment:
Can we always return a `guard` cursor?  (but change the name) and it just happens that the two cursors here coincide when there is no guard?